app/src/main/java/app/foodfinderapp/MainActivity.kt

```diff
@@ -90,25 +91,33 @@ class MainActivity : AppCompatActivity() {
             }
         }
 
-        val restaurantRef = db.collection("restaurants")
-        restaurantRef.get()
-            .addOnSuccessListener { documents ->
-            for (document in documents) {
-                val restaurant = document.toObject(Restaurant::class.java)
-                restaurantList.add(restaurant)
+        restaurantService.getAllRestaurants(
+            onSuccess = { fetchedRestaurants ->
+                restaurantList.addAll(fetchedRestaurants)
+                adapter.notifyDataSetChanged()
+            },
+            onFailure = { exception ->
+                Log.e(TAG, "Error getting restaurants", exception)
             }
-            adapter.notifyDataSetChanged()
-        }.addOnFailureListener { exception ->
-            Log.e(TAG, "Error getting restaurants", exception)
-        }
-
+        )
     }
 
+    /**
+     * Determines if the user is logged in.
+     *
+     * @return true if the user is logged in, returns false otherwise.
+     */
     fun isLoggedIn(): Boolean {
         val currentUser = FirebaseAuth.getInstance().currentUser
         return currentUser != null
     }
 
+    /**
+     * Initialize the contents of the Activity's standard options menu.
+     *
+     * @param menu The options menu in which to place the items.
+     * @return true for the menu to be displayed; if false, it will not be shown.
+     */
```

Comment:
These are overridden methods, and the original method signature contains JavaDoc.

If that's the case with a method that is overridden, especially a well known one, it's not always necessary to create new KDoc/JavaDoc above the overridden method.